### PR TITLE
use webpack to create a browser-ready version of the library

### LINKED ISSUE #157: Library causes errors in webpack
### Expected Behaviour

Using the library in a browser/webpack environment should be seamless.

### Actual Behaviour

Trying to use this library in a consumer using webpack causes an error due to the client's use of various node.js-only modules.

### Reproduce Scenario (including but not limited to)

Attempting to use the library in a React application.

#### Platform and Version

MacOS, module version 3.4.1, node.js version 22.

### PR DESCRIPTION
**Pull request by mfrisbey**
## Description

Updates the library's `es2015` build process so that it uses `webpack` instead of `babel` directly. Introduced a basic webpack configuration that uses polyfills for various node.js-only modules required by the library.

Also updates various dependencies to more secure versions, and changes the CI so that it runs tests on non-deprecated versions of node.js.

Note that I'll be releasing this as a breaking change, since the supported versions of node.js have changed, and because the output in the `es2015` build directory has changed.

Fixes #157 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by pheenomenon** [comment]
The ITs & dependent tests are failing. Likely unrelated.

**Review by pheenomenon — APPROVED** [review]

**Comment by tcallsen** [comment]
Ok, I've retried the ITs a couple times now.

**Comment by mfrisbey** [comment]
It's hard to parse the circleci output, but for the `dependent_module_tests`, I see things like this:

```
error: Could not init action on container (POST http://0.0.0.0:32790/init: responded with error: Initialization has failed due to: Error: Cannot find module 'node:os'
Require stack:
- /nodejsAction/gQkgSym7/index.js
- /nodejsAction/runner.js
- /nodejsAction/src/service.js
```

For the `integration_tests`, it looks like tests are passing, but the output ends with:

```
Too long with no output (exceeded 10m0s): context deadline exceeded
```

@pheenomenon Are we ok to merge despite the errors?

**Comment by pheenomenon** [comment]
I have created a jira to look more into it.. My initial analysis is, it's unrelated, as other open source CI had the same issue in the CI https://github.com/adobe/asset-compute-events-client/pull/36. 
Will take more analysis to figure out what's going on. If the merge is time critical,  think it should be ok to merge.

**Comment by mfrisbey** [comment]
@pheenomenon ok, thanks for looking at it. I'm planning on merging it as a breaking change, so it'll update the major version number of the library. Consumers will need to explicitly update the version, so if it ends up breaking anything it should be pretty easy to just revert while we figure things out.

I'll merge it for now, and we can revisit if there are issues. I did run all the unit tests and e2e tests locally, and they all passed. So hopefully risk is low.
